runtime/tests/runtime_integration_tests/integration_tests.rs

```diff
@@ -88,6 +88,72 @@ fn test_update() {
     assert_eq!(fw_rev[1], 0xaabbccdd);
 }
 
+///This test is planned to be executed only on nightly runs because of its long time to finish.
+#[cfg_attr(not(feature = "slow_tests"), ignore)]
+#[test]
+fn test_stress_update() {
+    let app_versions: [u32; 2] = [0xaaabbbbc, 0xaaabbbbd];
+
+    let image_options_0 = ImageOptions {
+        app_version: app_versions[0],
+        ..Default::default()
+    };
+
+    let image_options_1 = ImageOptions {
+        app_version: app_versions[1],
+        ..Default::default()
+    };
+
+    let image = [
+        caliptra_builder::build_and_sign_image(&FMC_WITH_UART, &APP_WITH_UART, image_options_0)
+            .unwrap()
+            .to_bytes()
+            .unwrap(),
+        caliptra_builder::build_and_sign_image(&FMC_WITH_UART, &APP_WITH_UART, image_options_1)
+            .unwrap()
+            .to_bytes()
+            .unwrap(),
+    ];
+    let mut model = run_rt_test(None, None, None);
+
+    let stress_num: u32 = 500;
+    let mut image_select = 0;
+
+    for _ in 0..stress_num {
+        model.step_until(|m| m.soc_mbox().status().read().mbox_fsm_ps().mbox_idle());
+
+        if image_select == 0 {
+            model
+                .mailbox_execute(u32::from(CommandId::FIRMWARE_LOAD), &image[image_select])
+                .unwrap();
+
+            model
+                .step_until_output_contains("Caliptra RT listening for mailbox commands...")
+                .unwrap();
+
+            //Check if the new firmware is actually the one we built
+            let fw_rev = model.soc_ifc().cptra_fw_rev_id().read();
+            assert_eq!(fw_rev[0], 0xaaaaaaaa);
+            assert_eq!(fw_rev[1], app_versions[image_select]);
+            image_select += 1;
+        } else {
+            model
+                .mailbox_execute(u32::from(CommandId::FIRMWARE_LOAD), &image[image_select])
+                .unwrap();
+
+            model
+                .step_until_output_contains("Caliptra RT listening for mailbox commands...")
+                .unwrap();
+
+            //Check if the new firmware is actually the one we built
+            let fw_rev = model.soc_ifc().cptra_fw_rev_id().read();
+            assert_eq!(fw_rev[0], 0xaaaaaaaa);
+            assert_eq!(fw_rev[1], app_versions[image_select]);
+            image_select -= 1;
+        }
```

Comment:
You should be able to simplify this quite a bit

```rust
if image_select == 0 {
    image_select = 1;
} else {
    image_select = 0;
}

model
  .mailbox_execute(u32::from(CommandId::FIRMWARE_LOAD), &image[image_select])
  .unwrap();

model
  .step_until_output_contains("Caliptra RT listening for mailbox commands...")
  .unwrap();

//Check if the new firmware is actually the one we built
let fw_rev = model.soc_ifc().cptra_fw_rev_id().read();
assert_eq!(fw_rev[0], 0xaaaaaaaa);
assert_eq!(fw_rev[1], app_versions[image_select]);
```

Reply:
Looks good. Updated the code.